### DISCUSSION TITLE
Welcome to ksqldb-go Discussions!

### DISCUSSION THREAD

**Discussion by thmeitz** [thmeitz/ksqldb-go#17] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  Best Thomas

**Comment by JeonEunseoung** [comment]
how to resolve this error when executing PullQuery -> can't do request : Post "localhost:8080": unexpected EOF
???

**Reply by thmeitz** [reply]
it seems, that this is the place, where the error message comes from.

https://github.com/thmeitz/ksqldb-go/blob/bd56bb34b0658a82fb6d5359c1263a08a9bb9368/pull.go#L79

As stated in the Readme file [Create a ksqlDB Client](https://github.com/thmeitz/ksqldb-go/blob/bd56bb34b0658a82fb6d5359c1263a08a9bb9368/README.md#create-a-ksqldb-client), you have to create a ksqldb client.

If you are using the docker-compose.yml from the example, the port is not 8080 => the default port is 8088

This is my cobra example

https://github.com/thmeitz/ksqldb-go/blob/bd56bb34b0658a82fb6d5359c1263a08a9bb9368/examples/cobra-test/cmd/root.go#L59

Hope that helps.

**Reply by JeonEunseoung** [reply]
I already made a ksqlDB client and topic, streams.
And I even checked that there was no error.
this is my source code.

``
func main() {
   options := net.Options{
      BaseUrl:   "http://myksqlDBaddress:myport",
      AllowHTTP: true,
   }

   kcl, err := ksqldb.NewClientWithOptions(options)
   if err != nil {
      fmt.Println("client fail")
   }

   defer kcl.Close()
   kcl.EnableParseSQL(true)
   
   _, r, err := kcl.Pull(context.TODO(), ksqldb.QueryOptions{Sql: "select * from table;"})

   if err != nil {
      fmt.Println(err)
   }

   var answers string
   for _, row := range r {
      if row != nil {
         answers = row[0].(string)
         fmt.Println(answers)
      }
   }

}
``

what should I do?

**Reply by thmeitz** [reply]
Could you please post the Docker Compose file that you use to run Kafka and KSQLDB? Or what does your runtime environment look like? The KSQLDB client is simply an HTTP client. I don't know if you have looked at my example: examples/cobra-test/cmd/pull.go

I can't help you if I don't know more about your Kafka + KSQLDB environment. The important thing is that the KSQLDB API port and the port in in your program "http://myksqlDBaddress:myport" must match. I also don't know if your Kafka is behind a firewall or or....
